pkg/deployer/baremetal/deployer.go

```diff
@@ -173,9 +175,62 @@ func (d *Deployer) CreateEtcdCluster(ctx context.Context, clusterName string, op
 		return err
 	}
 
+	if err := d.checkEtcdHealth(bin); err != nil {
+		return err
+	}
+
 	return nil
 }
 
+func (d *Deployer) checkEtcdHealth(etcdBin string) error {
+	// It's very likely that "etcdctl" is under the same directory of "etcd".
+	etcdctlBin := path.Join(etcdBin, "../etcdctl")
+	exists, err := utils.FileExists(etcdctlBin)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		d.logger.V(3).Infof("'etcdctl' is not found under the same directory of 'etcd', skip checking the healthy of Etcd.")
+		return nil
+	}
+
+	for retry := 0; retry < 10; retry++ {
```

Comment:
question: Can we use `etcdctl --endpoints=$ENDPOINTS endpoint health`([link](https://etcd.io/docs/v3.5/tutorials/how-to-check-cluster-status/))? It looks simpler.

Reply:
We need to make sure the Etcd do have a leader. I don't know if `health` check can be sure about that. Seems there are no "leader" info in it.